### DISCUSSION TITLE
How to make custom_lint warnings as errors to prevent project build?

### DISCUSSION THREAD

**Discussion by komalkhatri08** [invertase/.github#1] [Q&A]
How to make custom_lint warnings as errors to prevent project build? 
Even if I show it as warning and project is build successfully, how I can prevent it  while git commit? 
Eg. Script for pre-commit runs 'flutter analyze' and prevents from commiting the code.

**Comment by rrousselGit** [comment]
Update your pre-commit script to also run `dart run custom_lint`

**Reply by komalkhatri08** [reply]
```
#!/bin/bash

# Run custom_lint rules
echo "Running custom_lint..."
dart run custom_lint

# Check the exit code of custom_lint
if [ $? -ne 0 ]; then
  echo "Custom linting issues found. Aborting commit."
  exit 1
fi

# Run flutter analyze
echo "Running flutter analyze..."
flutter analyze

# Check the exit code of flutter analyze
if [ $? -ne 0 ]; then
  echo "Linting issues found. Aborting commit."
  exit 1
fi 
```
I m writing like this. It is not working . Says 'could not find custom_lint'

**Reply by rrousselGit** [reply]
When using `dart run`, you need to be in a Dart project.

Otherwise you'll have to install custom_lint globally (`dart pub global activate custom_lint`) and run `custom_lint` (without `dart run`)

**Reply by komalkhatri08** [reply]
Running custom_lint...
.git/hooks/pre-commit: line 6: custom_lint: command not found
Custom linting issues found. Aborting commit.


git did not exit cleanly (exit code 1) (1141 ms @ 8/31/2023 4:35:25 PM)

This is error which I get while code commit. I followed above steps suggested by you.

**Reply by komalkhatri08** [reply]
@rrousselGit  , do u have some comments on this ? It would be real help if we sort this out. Thanks! :)
